### PR TITLE
Remove emotion dependency and fix peer dependency warnings

### LINKED ISSUE #9857: Remove emotion package
Prompted by https://github.com/mozilla/addons-frontend/pull/9853, I was looking to see what we use the `emotion` package for, and I cannot find any references to it. Locally, if I remove it I don't seem to have any problems. The build doesn't break, the site works fine, and Storybook works fine.

@willdurand Do you think we can remove it?

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #9857 

Interestingly, even if I remove these from `package.json`, when running `yarn` they are both added into the `yarn.lock` file, so I guess they are dependencies of something else?

If we don't need them as explicit dependencies, it seems worth removing them, no?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -271,7 +271,6 @@
     "@babel/preset-flow": "^7.0.0",
     "@babel/preset-react": "^7.0.0",
     "@babel/register": "^7.0.0",
-    "@emotion/core": "^10.0.16",
```
Comment (willdurand) [inline package.json:274]:
```
warning "react-storybook-addon-chapters > @emotion/styled@10.0.27" has unmet peer dependency "@emotion/core@^10.0.27"
```

**Review by willdurand — CHANGES_REQUESTED** [review]
We need to add `@emoticon/core` back.

File: package.json (hunk 1 above)
Comment (bobsilverberg) [reply package.json:274]: We have a number of those warnings which have been present for ages. I know at one point we got rid of all of them, but it seems that for a long time we've just been ignoring them, and it doesn't seem to cause any issues. Are you suggesting that we should look at each one and try to get rid of it?

Comment (willdurand) [reply package.json:274]: We could take a look at the other ones but I don't see the point of merging a patch that will add a new warning there :) Removing the other dependency should be fine, though.

Comment (bobsilverberg) [reply package.json]: I found a couple of quick wins in terms of peer dependency warnings, so I added those in as well.

**Review by willdurand — APPROVED** [review]
Thanks!

Quick question: no lockfile changes?

**Comment by bobsilverberg** [comment]
> Thanks!
> 
> Quick question: no lockfile changes?

There were some, but I've run into problems in the past accepting local lockfile changes and adding them to a commit, so I've found it better to just exclude it and allow renovate to update it as it sees fit.  Having said that, I'll take a look at the diff for the lockfile and see if it looks safe to include.

**Comment by willdurand** [comment]
We definitely need the lockfile changes (all of them).

**Comment by bobsilverberg** [comment]
> We definitely need the lockfile changes (all of them).

I've added it to the patch.

**Pull request merged**